src/apify/storages/request_list.py

```diff
@@ -0,0 +1,137 @@
+    r'https?:\/\/(www\.)?([^\W_]|[^\W_][-\w0-9@:%._+~#=]{0,254}[^\W_])\.[a-z]{2,63}(:\d{1,5})?(\/[-\w@:%+.~#?&/=()]*)?'
+)
+
+
+class _RequestDetails(BaseModel):
+    method: HttpMethod = 'GET'
+    payload: str = ''
+    headers: dict[str, str] = Field(default_factory=dict)
+    user_data: dict[str, str] = Field(default_factory=dict, alias='userData')
+
+
+class _RequestsFromUrlInput(_RequestDetails):
+    requests_from_url: str = Field(alias='requestsFromUrl')
+
+
+class _SimpleUrlInput(_RequestDetails):
+    url: str
+
+
+url_input_adapter = TypeAdapter(list[Union[_RequestsFromUrlInput, _SimpleUrlInput]])
+
+
+class RequestList(CrawleeRequestList):
+    """Extends crawlee RequestList."""
+
+    @classmethod
+    async def open(
+        cls,
+        name: str | None = None,
+        actor_start_urls_input: list[dict[str, Any]] | None = None,
+        http_client: BaseHttpClient | None = None,
+    ) -> RequestList:
+        """Creates RequestList from Actor input requestListSources.
+
+        name is name of the returned RequestList
+        actor_start_urls_input  can contain list dicts with either url or requestsFromUrl key
+        http_client is client that will be used to send get request to url defined in requestsFromUrl
+
+        Example actor_start_urls_input:
+            [
+                # Gather urls from response body.
+                {'requestsFromUrl': 'https://crawlee.dev/file.txt', 'method': 'GET'},
+                # Directly include this url.
+                {'url': 'https://crawlee.dev', 'method': 'GET'}
+            ]
+        """
+        actor_start_urls_input = actor_start_urls_input or []
+        return await _create_request_list(name, actor_start_urls_input, http_client)
+
+
+async def _create_request_list(
+    name: str | None, actor_start_urls_input: list[dict[str, Any]], http_client: BaseHttpClient | None
+) -> RequestList:
+    if not http_client:
+        http_client = HttpxHttpClient()
+
+    ulr_inputs = url_input_adapter.validate_python(actor_start_urls_input)  # instance of list[Union[...]]
+
+    simple_url_inputs = [url_input for url_input in ulr_inputs if type(url_input) is _SimpleUrlInput]
+    remote_url_inputs = [url_input for url_input in ulr_inputs if type(url_input) is _RequestsFromUrlInput]
+
+    simple_url_requests = _create_requests_from_input(simple_url_inputs)
+    remote_url_requests = await _create_requests_from_url(remote_url_inputs, http_client=http_client)
+
+    return RequestList(name=name, requests=simple_url_requests + remote_url_requests)
+
+
+def _create_requests_from_input(simple_url_inputs: list[_SimpleUrlInput]) -> list[Request]:
+    return [
+        Request.from_url(
+            method=request_input.method,
+            url=request_input.url,
+            payload=request_input.payload.encode('utf-8'),
+            headers=request_input.headers,
+            user_data=request_input.user_data,
+        )
+        for request_input in simple_url_inputs
```

Comment:
Btw thinking about this class reminded me of this: https://github.com/apify/crawlee/pull/2466

Basically, in JS Crawlee, we used to instantiate all the requests at the beginning (like here), but that ate up much more memory than just keeping the POJO objects (I guess `dict`s in Python) and instantiating the `Request` class only once we really need it. Maybe Python handles this better (has slimmer class instances).

It was at that time we learned that users love to make >1.000.000 requests long RLs 😅

Reply:
That is a valid point, but I think it will be done later in https://github.com/apify/crawlee-python/issues/99 as the current class is openly not as mature as the JS version:

https://github.com/apify/crawlee-python/blob/master/src/crawlee/storages/_request_list.py#L23

Reply:
Agreed, let's make sure the SDK supports everything the platform can offer first, then we can optimize.